kolibri/logger/management/commands/generateuserdata.py

```diff
@@ -30,307 +23,61 @@ def add_arguments(self, parser):
-                                        'completion_timestamp': completion_timestamp,
-                                        'mastery_criterion': {
-                                            'm': 5,
-                                            'n': 5,
-                                            'type': 'm_of_n',
-                                        },
-                                    }
-                                )
-
-                                if not created:
-                                    # Not created, so update relevant fields on it based on new interactions
-                                    if not mastery_log.complete and summary_log.progress >= 1.0:
-                                        mastery_log.complete = True
-                                        mastery_log.completion_timestamp = summary_log.completion_timestamp
-                                    mastery_log.end_timestamp = summary_log.end_timestamp
-
-                                # Get the list of assessment item ids from the assessment meta data
-                                assessment_item_ids = random_node.assessmentmetadata.first().assessment_item_ids
-
-                                for i, session_log in enumerate(reversed(session_logs)):
-                                    # Always make students get 5 attempts correct in the most recent session
-                                    # if the exercise is complete
-                                    complete = (i == 0 and mastery_log.complete)
-                                    if complete:
-                                        n = 5
-                                    else:
-                                        # Otherwise, let them have answered between 1 and 5 questions per session
-                                        n = random.randint(1, 5)
-                                    # How long did they spend on these n questions?
-                                    timespan = session_log.end_timestamp - session_log.start_timestamp
-                                    # Index through each individual question
-                                    for j in range(0, n):
-                                        if complete:
-                                            # If this is the session where they completed the exercise, always
-                                            # make them get it right
-                                            correct = True
-                                        else:
-                                            # Otherwise only let students get odd indexed questions right,
-                                            # ensuring they will always have a mastery breaking sequence
-                                            # as zero based indexing means their first attempt will always be wrong!
-                                            correct = j % 2 == 1
```

Comment:
Amazing, didn't this code fail the linting? McCabe complexity level seems extremely high!!

Reply:
That's why there was a `#  noqa` at the top. I know, because I copied it from where you did the same thing somewhere else :)

Reply:
Hahaha, *blushes*